synapse/federation/federation_server.py

```diff
@@ -607,3 +565,66 @@ def on_exchange_third_party_invite_request(self, origin, room_id, event_dict):
             origin, room_id, event_dict
         )
         defer.returnValue(ret)
+
+
+class FederationHandlerRegistry(object):
+    """Allows classes to register themselves as handlers for a given EDU or
+    query type for incoming federation traffic.
+    """
+    def __init__(self):
+        self.edu_handlers = {}
+        self.query_handlers = {}
+
+    def register_edu_handler(self, edu_type, handler):
+        """Sets the handler callable that will be used to handle an incoming
+        federation EDU of the given type.
+
+        Args:
+            edu_type (str): The type of the incoming EDU to register handler for
+            handler (Callable[str, dict]): A callable invoked on incoming EDU
```

Comment:
I think `Callable[str, dict]` means "a thing that takes a str and returns a dict". You probably want `Callable[[str, dict], Deferred[None]]` or something?

Reply:
Oops, I misread the thingy. Ta.